### PR TITLE
Last new simulator rpc's 

### PR DESCRIPTION
**Pull request by jack60612**
Pretty self explanatory pr, tests coming soon.
Does not explicitly require #12521 but it would be nice.
- [x] Coin store addition, get_all_coins & get_all_puzzle_hashes RPC
- [x] Chain Reversion RPC (complete block deletion  & reorg RPC's)
- [x] Unit & Integration Tests (all 4 rpc's.)

### CODE REVIEW

**Comment by jack60612** [comment]
Tests were added, PR is now ready for review.

**Comment by jack60612** [comment]
Slight Sql change made, still ready for review.

**Review by trepca — COMMENTED** [review]
why are some file names in camel case? (this is not Java :) )

**Comment by jack60612** [comment]
> why are some file names in camel case? (this is not Java :) )

The other protocols we used for the simulator did the same, so I was just following what we already did.

File: chia/full_node/coin_store.py [hunk 1]
```diff
@@ -242,6 +242,22 @@ async def get_coins_removed_at_height(self, height: uint32) -> List[CoinRecord]:
                         coins.append(coin_record)
                 return coins
 
+    async def get_all_coins(self, include_spent_coins: bool) -> List[CoinRecord]:
+        # this is used by the simulator, using this normally would be a joke.
```
Comment (trepca) [inline chia/full_node/coin_store.py:246]: can we elaborate this a bit like `WARNING: this should only be used for testing or simulation, running it on a synced testnet or mainnet node will most likely result in OOM`

File: chia/simulator/full_node_simulator.py [hunk 2]
```diff
@@ -69,6 +70,45 @@ async def update_autofarm_config(self, enable_autofarm: bool) -> bool:
                 await self.farm_new_transaction_block(FarmNewBlockProtocol(self.bt.farmer_ph))
             return self.auto_farm
 
+    @api_request
+    async def get_all_coins(self, request: GetAllCoinsProtocol) -> List[CoinRecord]:
+        return await self.full_node.coin_store.get_all_coins(request.include_spent_coins)
+
+    async def revert_block_height(self, new_height: uint32) -> None:
+        # this completely deletes blocks from the blockchain.
```
Comment (trepca) [inline chia/simulator/full_node_simulator.py:78]: best if we use [docstring](https://peps.python.org/pep-0257/) for this

File: chia/full_node/coin_store.py (hunk 1 above)
Comment (jack60612) [reply chia/full_node/coin_store.py:246]: yeah that would probably be better :)

**Review by trepca — APPROVED** [review]

**Comment by jack60612** [comment]
🎺

**Pull request merged**